tests/test_embedded_scheduler.cpp

```diff
@@ -323,6 +333,54 @@ TEST(TestEmbeddedScheduler, EventLoopPoll)
     EXPECT_THAT(evl.getTree()[0U]->getDeadline().value().time_since_epoch(), 210ms);  // Skipped ahead!
 }
 
+TEST(TestEmbeddedScheduler, EventLoopDefer_single_overdue)
+{
+    using time_point = SteadyClockMock::time_point;
+    using std::chrono_literals::operator""ms;
+    SteadyClockMock::reset();
+    EventLoop<SteadyClockMock> evl;
+
+    auto evt = evl.defer(SteadyClockMock::now() + 1000ms, [&](auto) {});
+    EXPECT_TRUE(evt);
+
+    // This is special case - only one deferred event (and no "repeat"-s!), and it is already overdue (by +30ms).
+    // So, `next_deadline` should be `time_point::max()` b/c there will be nothing left pending after spin.
+    SteadyClockMock::advance(1000ms + 30ms);
+    const auto out = evl.spin();
+    EXPECT_THAT(out.next_deadline.time_since_epoch(), time_point::max().time_since_epoch());
```

Comment:
At this line @ 350 we've got incorrect next deadline time (1000ms instead of correct max).